// Copyright (c) 2010-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

#ifndef OPENTXS_PROTOBUF_ASYMMETRICKEY_HPP
#define OPENTXS_PROTOBUF_ASYMMETRICKEY_HPP

#include "opentxs/Version.hpp"  // IWYU pragma: associated

#include "opentxs/protobuf/Enums.pb.h"

namespace opentxs
{
namespace proto
{
class AsymmetricKey;
}  // namespace proto
}  // namespace opentxs

namespace opentxs
{
namespace proto
{
auto CheckProto_1(
    const AsymmetricKey& key,
    const bool silent,
    const CredentialType type,
    const KeyMode mode,
    const KeyRole role) -> bool;
auto CheckProto_2(
    const AsymmetricKey&,
    const bool,
    const CredentialType,
    const KeyMode,
    const KeyRole) -> bool;
auto CheckProto_3(
    const AsymmetricKey&,
    const bool,
    const CredentialType,
    const KeyMode,
    const KeyRole) -> bool;
auto CheckProto_4(
    const AsymmetricKey&,
    const bool,
    const CredentialType,
    const KeyMode,
    const KeyRole) -> bool;
auto CheckProto_5(
    const AsymmetricKey&,
    const bool,
    const CredentialType,
    const KeyMode,
    const KeyRole) -> bool;
auto CheckProto_6(
    const AsymmetricKey&,
    const bool,
    const CredentialType,
    const KeyMode,
    const KeyRole) -> bool;
auto CheckProto_7(
    const AsymmetricKey&,
    const bool,
    const CredentialType,
    const KeyMode,
    const KeyRole) -> bool;
auto CheckProto_8(
    const AsymmetricKey&,
    const bool,
    const CredentialType,
    const KeyMode,
    const KeyRole) -> bool;
auto CheckProto_9(
    const AsymmetricKey&,
    const bool,
    const CredentialType,
    const KeyMode,
    const KeyRole) -> bool;
auto CheckProto_10(
    const AsymmetricKey&,
    const bool,
    const CredentialType,
    const KeyMode,
    const KeyRole) -> bool;
auto CheckProto_11(
    const AsymmetricKey& key,
    const bool silent,
    const CredentialType type,
    const KeyMode mode,
    const KeyRole role) -> bool;
auto CheckProto_12(
    const AsymmetricKey&,
    const bool,
    const CredentialType,
    const KeyMode,
    const KeyRole) -> bool;
auto CheckProto_13(
    const AsymmetricKey&,
    const bool,
    const CredentialType,
    const KeyMode,
    const KeyRole) -> bool;
auto CheckProto_14(
    const AsymmetricKey&,
    const bool,
    const CredentialType,
    const KeyMode,
    const KeyRole) -> bool;
auto CheckProto_15(
    const AsymmetricKey&,
    const bool,
    const CredentialType,
    const KeyMode,
    const KeyRole) -> bool;
auto CheckProto_16(
    const AsymmetricKey&,
    const bool,
    const CredentialType,
    const KeyMode,
    const KeyRole) -> bool;
auto CheckProto_17(
    const AsymmetricKey&,
    const bool,
    const CredentialType,
    const KeyMode,
    const KeyRole) -> bool;
auto CheckProto_18(
    const AsymmetricKey&,
    const bool,
    const CredentialType,
    const KeyMode,
    const KeyRole) -> bool;
auto CheckProto_19(
    const AsymmetricKey&,
    const bool,
    const CredentialType,
    const KeyMode,
    const KeyRole) -> bool;
auto CheckProto_20(
    const AsymmetricKey&,
    const bool,
    const CredentialType,
    const KeyMode,
    const KeyRole) -> bool;
}  // namespace proto
}  // namespace opentxs

#endif  // OPENTXS_PROTOBUF_ASYMMETRICKEY_HPP
